etna/ciim/client.py

```diff
@@ -134,6 +152,64 @@ def prepare_filter_aggregations(items: Optional[list]) -> Optional[str]:
     return filter_prepared_list
 
 
+class ResultList:
+    """
+    Wrapper class for JSON-decoded list of search results, that lazily converts
+    raw "hits" into instances of `item_type` when iterated, as well
+    providing the following helper attributes for a more developer-friendly
+    interface:
+
+    `total_count`:
+        The total number of results available
+
+    `aggregations`:
+        A dict of "aggregation" values from the results themselves
+
+    `filter_aggregations`:
+        A dict of "aggregation" values populated from the separate 'filter aggregations'
+        response from the API. For endpoints that do not include such data in their
+        response (e.g. 'searchAll/' and 'searchUnified/'), the value will be `None`.
```

Comment:
@ababic There is a separate handling and meaning of filter_aggregations - its as search parameter `filterAggregations` . It looks **confusing** as the value assigned appears to be from a api response. Please look at this along with the comment of using `filter_aggregations_response` in `KongClient.search`. I think its not required, but other parts may be impacted.

Reply:
The `filter_aggregations` attribute has now been removed, and the `bucket_counts` attribute added in its place.

For the 'search' endpoint, this value comes from `["aggregations"]["groups"]["buckets"]` in first ES response, but in all other cases, it is taken from the same response that contains `["hits"]`